core/src/main/java/com/vml/es/aem/acm/core/code/Condition.java

```diff
@@ -303,6 +305,30 @@ public LocalTime dayEndTime() {
         return LocalTime.of(23, 59, 59, 999999999);
     }
 
+    // Date based
+
+    public boolean isDate(String dateString) {
+        LocalDateTime localDateTime = DateUtils.localDateTimeFromString(dateString);
+        return isDate(localDateTime);
+    }
+
+    public boolean isDate(ZonedDateTime zonedDateTime) {
+        LocalDateTime localDateTime =
+                zonedDateTime.withZoneSameLocal(ZoneOffset.systemDefault()).toLocalDateTime();
+        return isDate(localDateTime);
+    }
+
+    public boolean isDate(LocalDateTime localDateTime) {
+        long interval = executionContext
+                .getOsgiContext()
+                .getService(ScriptScheduler.class)
+                .getIntervalBetweenRuns();
+        if (interval < 0) {
+            throw new IllegalArgumentException("Invalid cron expression in scheduler_expression OSGi config");
```

Comment:
log.error inside script scheduler; do not expose internal details in human readable messages
 even I am not sure if that validation is needed at all